RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -23,20 +23,38 @@ class PFTICLProducer : public edm::global::EDProducer<> {
   void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
 
 private:
+  // parameters
+  const bool useTimingQuality_, useTimingAverage_;
+  const float timingQualityThreshold_;
+
   // inputs
   const edm::EDGetTokenT<edm::View<TICLCandidate>> ticl_candidates_;
+  const edm::EDGetTokenT<edm::ValueMap<float>> srcTrackTime_, srcTrackTimeError_, srcTrackTimeQuality_;
 };
 
 DEFINE_FWK_MODULE(PFTICLProducer);
 
 PFTICLProducer::PFTICLProducer(const edm::ParameterSet& conf)
-    : ticl_candidates_(consumes<edm::View<TICLCandidate>>(conf.getParameter<edm::InputTag>("ticlCandidateSrc"))) {
+    : useTimingQuality_(conf.existsAs<edm::InputTag>("trackTimeQualityMap")),
+      useTimingAverage_(conf.existsAs<bool>("useTimingAverage") ? conf.getParameter<bool>("useTimingAverage") : false),
```

Comment:
No need to check `existsAs`, because it is defined in the `filDescriptions` method

Reply:
Done:
https://github.com/cms-sw/cmssw/pull/32046/commits/65e52dd8eb097b00b185bf1f106e9c87ea99e053#diff-49ac77e92d14b3c79dfe08879ad1b7f4054eb34bf175e844010e4a77c0d242f1R38